pkg/controller/sparkapplication/controller.go

```diff
@@ -672,9 +683,54 @@ func (c *Controller) submitSparkApplication(app *v1beta1.SparkApplication) *v1be
 			}
 		}
 	}
+
+	//Use PodGroup to schedule spark application executor pods if required
+	if c.volcanoClient != nil && app.VolcanoSchedulingRequired() {
+		if app.Spec.Executor.Annotations == nil {
+			app.Spec.Executor.Annotations = make(map[string]string)
+		}
+		if _, ok := app.Spec.Executor.Annotations["scheduling.k8s.io/group-name"]; !ok {
+			if pgName := c.createPodGroupIfNeeded(app); pgName != "" {
+				app.Spec.Executor.Annotations["scheduling.k8s.io/group-name"] = pgName
+			}
+		}
+	}
 	return app
 }
 
+func (c *Controller) createPodGroupIfNeeded(app *v1beta1.SparkApplication) string {
```

Comment:
How about return a `(string, error)` instead and change `glog.Warningf` below to `return "", fmt.Errorf(...)`, and then let's the caller handle the error?